cmd/limactl/hostagent.go

```diff
@@ -105,7 +105,7 @@ func hostagentAction(cmd *cobra.Command, args []string) error {
 	go func() {
 		defer os.RemoveAll(socket)
 		defer srv.Close()
-		if serveErr := srv.Serve(l); serveErr != nil {
+		if serveErr := srv.Serve(l); serveErr != http.ErrServerClosed {
```

Comment:
Where would we ever call `Close` or `Shutdown` on the server? Will it not just continue to run until `l.Accept()` returns an error, and then the error from `srv.Serve()` will be just that?

Unless I'm mistaken the new condition will never be true (just like the old one), but at least it is somewhat "more correct".

Practically it could also be just

```go
serveErr := srv.Serve(l)
logrus.WithError(serveErr).Warn("hostagent API server exited")
```

Reply:
I'm looking on this.